docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -735,28 +563,86 @@ As database IDs are all globally unique, it's fine to replace table ID with data
+  - The DDL statement to `ALTER TABLE t1 ADD PLACEMENT` has been removed from the proposal.
+  - The DDL statement `CREATE PLACEMENT POLICY` has been added (allowing common configurations to be saved).
+  - Configuring placement rules for indexes is no longer supported (we can add it once global indexes are added).
+  - The inheritance rules have been simplified to match character set/collation inheritance.
+2. There is a risk that we do not fully understand complaince requirements (see unresolved questions), or that the substaintial effort required to achieve compliance with DDL, internal SQL and DML.
+3. Related to the above point, there is risk that the implementation of compliance requirements has significant burden on multiple teams, including ecosystem tools (Backup, CDC, DM). Even tools such as dumpling should ideally be able to backup placement rules in logical form.
+4. There is some risk that a common use case can not be expressed in `CONSTRAINT` syntax, leading to complicated scenarios where users still need to express placement by using PD directly. Ideally, we can recommend users use SQL rules exclusively.
+5. Many other features in TiDB are in development, some of which may influence placement rules. Clustered index affects the key format of primary index. Fortunately, the prefix of key range is untouched. Global secondary index largely affect the placement rules of partitioned tables. The key range of one global secondary index is not successive, so if it's necessary to define placement rules on the index, multiple rules should be generated in the PD. But for now, there's no such scenario.
+
+
+## Investigation & Alternatives
+
+For investigation, we looked at the implementation of placement rules in various datases (CockroachDB, Yugabyte, OceanBase).
+
+The idea of using a `PLACEMENT POLICY` was inspired by how OceanBase has Placement Groups, which are then applied to tables. But the usage as proposed here is optional, which allows for more flexibility for casual cases. The idea of using a Placement Group can also be seen as similar to using a "tablespace" in a traditional database, but it's not completely the same since the choice is less binary (constraints allow the placement of roles for leaders, followers, learners etc.)
+
+CockroachDB does not look to have something directly comparable to `PLACEMENT POLICY`, but it does have the ability to specify "replication zones" for "system ranges" such as default, meta, liveness, system, timeseries. Before dropping `ALTER TABLE t1 ADD PLACEMENT` from this proposal, it was investigated the CockroachDB does not support this syntax, presumably for simplification and minimising similar risks of misconfiguration.
+
+## Unresolved Questions
```

Comment:
I proposed a new syntax yesterday in the lurk after djshow832 summarized the meeting:

> IMO, the only bad part is that, I need to write twice "[+region=A,+zone=1] leader, [+region=A,+zone=1] follower" to configure region#A zone#1, while it can be expressed in one line "In region A  zone #1,  give me 1 leader and 1 follower".
>Maybe we can split the settings of CONSTRAINTS and other settings, .e.g.:
>1. Set replicas in another statement(syntactically) `ALTER TABLE ... SET CONSTRAINTS=xxx, SET FOLLOWER=3`.
>2. One constraints with multiple roles, `ALTER TABLE ...SET CONSTRAINTS=xxx LEADER=1 FOLLOWER=1`.

I still think `CONSTRAINTS=[...] LEADER=1 FOLLOWER=1` will be a good alternative to the `{xxx:1, vvv:2}` syntax. Since we don't really have `ADD PLACEMENT/DROP PLACEMENT` based on roles, we can assign rules based on `CONSTRAINTS`.

Reply:
Thanks, I will look again tomorrow.

Reply:
This has been incorporated. Thank you for the suggestion.